src/protocol/http/org/apache/jmeter/protocol/http/control/CookieManager.java

```diff
@@ -428,7 +438,9 @@ public void testEnded(String host) {
     /** {@inheritDoc} */
     @Override
     public void testIterationStart(LoopIterationEvent event) {
-        if (getClearEachIteration()) {
+        JMeterVariables jMeterVariables = JMeterContextService.getContext().getVariables();
+        if ((getControlledByThread() && !jMeterVariables.isSameUserOnNextIteration()) 
+                || getClearEachIteration()) {
```

Comment:
Does that mean `getClearEachIteration` impacts the result no matter of `getControlledByThread`?

Reply:
Yes, but if you look at GUI it's explicit